internal/bminventory/inventory.go

```diff
@@ -633,7 +634,7 @@ func (b *bareMetalInventory) createAndUploadNodeIgnition(ctx context.Context, cl
 	if address == "" {
 		address = swag.StringValue(cluster.APIVipDNSName)
 	}
-	ignitionBytes, err := b.formatNodeIgnitionFile(address)
+	ignitionBytes, err := b.formatNodeIgnitionFile(address, host.MachineConfigPoolName)
```

Comment:
i think you need a default for config/worker in case user have not issued UpdateCluster with MachineConfigPool?

Reply:
The default is set on `updateRole` to match the Role every time.